apps/www/_blog/2024-07-01-supabase-realtime-broadcast-and-presence-authorization.mdx

```diff
@@ -0,0 +1,186 @@
+supabase.realtime.setAuth(token)
+
+supabase
+    .realtime
+    .channel("locked", { config: { private: true })
+    .subscribe((status: string, err: any) => {
+        status == "CHANNEL_ERROR" && console.error(err.message);
+    });
+```
+
+### More complex example
+
+You can find a more complex example in the [Supabase demo folder](https://github.com/supabase/supabase/tree/master/examples/realtime/nextjs-authorization-demo) where we are using this feature to build chat rooms with restricted access.
+
+## How does it work?
+
+We decided on an approach that keeps your database and RLS policies at the heart of the new authorization strategy.
+
+### Database as a source of security
+
+To achieve Realtime authorization, we looked into our current solutions, namely how [Storage](https://supabase.com/docs/guides/storage/security/access-control) handles Access Control. Due to the nature of Realtime, our primitives are different as we have no assets stored in the database. So how did we achieve it?
+
+We let you define that the channel itself is private! On Channel subscription you are able to inform Realtime to use a private channel and we will do the required checks.
+
+The checks are done by running SELECT and INSERT queries on the new `realtime.messages` table which are then rolled backed so nothing is persisted. Then, based on the query result, we can determine the policies the user has for a given extension.
+
+<Img
+  src={{
+    dark: '/images/blog/2024-07-01-supabase-realtime-broadcast-and-presence-authorization/schema--dark.png',
+    light:
+      '/images/blog/2024-07-01-supabase-realtime-broadcast-and-presence-authorization/schema--light.png',
+  }}
+  alt="The table that is used to test the policies."
+/>
+
+### One user, one context, one connection
+
+Now that we have set up everything on the database side, let's understand how it works and how we can verify authorization via RLS policies.
+
+Upon channel subscription, we read the contents of your configuration and determine that you want to use a private channel, then we parse the information from your signed JSON Web Token (JWT), load them into a Postgres transaction using `set_config`, and verify them against the new `realtime.messages`table within the same transaction and we store the output as a group of policies within the context of the user's channel process.
+
+<Img
+  src={{
+    dark: '/images/blog/2024-07-01-supabase-realtime-broadcast-and-presence-authorization/one-user-connection--dark.png',
+    light:
+      '/images/blog/2024-07-01-supabase-realtime-broadcast-and-presence-authorization/one-user-connection--light.png',
+  }}
+  alt="Flow of checks done against the databse to determine the policies for a user connection."
+/>
+
+Since we're doing this as soon as the user subscribes to Channel we should see a small increase in subscriptions but after that we do not run any queries on the database and we use the stored context to control the actions a user can perform.
```

Comment:
```suggestion
Since we're doing this as soon as the user subscribes to Channel we should see a small latency increase in subscriptions but after that we do not run any queries on the database and we use the stored context to control the actions a user can perform.
```

I think?